ISISScientificComputing/autoreduce
# Reconnect on disconnect AR-88 AR-1245 AR-1246 AR-1274

### Summary of work
The listener now implements the stomp listener base class as it should
The queue client now removes its listeners when it disconnects as it should
The queue client is refactored slightly as well as the listener
A timeout has been added to the wait part of the end to end test.
Adds a system test for activemq going down 

### How to test your work
Test should pass
Stop ActiveMQ, start ActiveMQ submit run, should still be picked up

### Additional comments
 

Fixes #xyz


**Before merging ensure the release notes have been updated**
